recipe/meta.yaml

```diff
@@ -1,5 +1,11 @@
 {% set version = "2.0.0" %}
 
+{% if cuda_compiler_version in (None, "None", True, False) %}
+{% set cuda_major = 0 %}
+{% else %}
+{% set cuda_major = environ.get("cuda_compiler_version", "11.8").split(".")[0] | int %}
+{% endif %}
```

Comment:
I find this somewhat awkward and overengineered (e.g. `cuda_compiler_version` is already available, why get it from `environ`?), but OK.

Reply:
Think this [comes from `cupy` originally]( https://github.com/conda-forge/cupy-feedstock/blob/11d783c60f134a359f8b617bf21cc748b8463c90/recipe/meta.yaml#L10-L14 ) and other feedstocks ([like `pytorch`]( https://github.com/conda-forge/pytorch-cpu-feedstock/blob/7c7a57b7515eaeda67d3879b56b68466f38f0b0d/recipe/meta.yaml#L15-L19 )) have been reusing it

Maybe there is a better alternative we can come up with (possibly by adding things to `zip_keys` or adding better Jinja functions for parsing versions

That said, for this case, I don't think we actually need this as the dependency below don't seem to be strictly needed

```suggestion
```

Reply:
I have been copy pasting this snippet in many of our feedstocks. I have not found a better way to differentiate between old ways of installing cuda previous to 12 and after than using the major version.

Reply:
Yeah think what you are doing is reasonable

Axel's comment just makes me think we should have better tooling so that maintainers (like yourself) can accomplish the same end result without needing this boilerplate